### ISSUE TITLE
thanos-store is too slow for query and don't know where the time went

### ISSUE THREAD

**Issue by Melody-zyy** [thanos-io/thanos#6945]
**Thanos, Prometheus and Golang version used**:
thanos-store version:v0.30.2

**Object Storage Provider**:
txyun cos

**What happened**:
i used thanos-store to query history data，but its too slow，i has collected many metrics, **but I still don't know where the time went**

**What you expected to happen**:
i want to data return in 20 seconds or i want to know where the time went, when i query hisotry data 

**How to reproduce it (as minimally and precisely as possible)**:

**Full logs to relevant components**:
This log is because the query reaches the thanos-query timeout, which interrupts subsequent thanos-store queries to the memcached cache

```
level=warn ts=2023-11-30T14:51:06.285175671Z caller=memcached_client.go:436 name=index-cache msg="failed to fetch items from memcached" numKeys=5 firstKey=P:01HFYJB6BXH9BD03Z6H29VXEXG:iWpYeQElI53MP3C4cUnj44BIr-dGk2AQ8I4-BjlebIg err="context canceled"
```

**Anything else we need to know**:

1. thanos-store has 10 shards based on the blocked id
2. thanos-compactor normal run in compact and downsample
3. i used memcached to cache
4. thanos-query --store.response-timeout=40s

**thanos-store config：**
![image](https://github.com/thanos-io/thanos/assets/73515600/01dc0a1c-dfc5-497d-b9d7-6da6c7042f5f)
**index-cache.yml：**
![image](https://github.com/thanos-io/thanos/assets/73515600/eeaef763-4029-4e8d-afdb-9b2d0bd24b46)
**bucket-cache.yml：**
![image](https://github.com/thanos-io/thanos/assets/73515600/0170e04d-28da-4398-9310-ed11b9fbfc5d)

**Comment by yeya24** [comment]
There could be different reasons. Do you have tracing setup? 
Or if you enable debug logging in Thanos store, we can take a look at `stats query processed` and there are detailed query stats

**Comment by Melody-zyy** [comment]
> There could be different reasons. Do you have tracing setup? Or if you enable debug logging in Thanos store, we can take a look at `stats query processed` and there are detailed query stats

I want to know whether the time is spent on data pulling or data processing, or if there is a problem with the memcached server? Are there any indicators to reflect that
